src/type/definition.js

```diff
@@ -102,6 +102,14 @@ export function isCompositeType(type: ?GraphQLType): boolean {
   );
 }
 
+export function typeHasFields(type: ?GraphQLType): boolean {
```

Comment:
That has different semantics. It does not include input types and includes unions